File: concept/13_char/palindrome/palindrome.cpp
#include <iostream>
using namespace std;

int getLength(char name[])
{
    int count = 0;
    for (int i = 0; name[i] != '\0'; i++)
    {

        count++;
    }
    return count;
}
bool isPalindrome(char name[], int length)
{
    int start = 0;
    int end = length - 1;
    while (start <= end)
    {
        if (name[start] != name[end])
        {
            return false;
        }
        else
        {
            start++;
            end--;
        }
    }
    return true;
}

int main()
{

    char name[10];

    cout << "Enter your name" << endl;
    cin >> name;

    int length = getLength(name);

    cout << length << endl;
    cout << isPalindrome(name, length) << endl;
}

File: concept/13_char/palindrome/palindrome.dart
import 'dart:io';

void main() {
  print("Enter value : ");
  String? name = stdin.readLineSync();

  print(name?.reversed);
  print(name?.isPalindrome);
}

bool isPalindromeString(String name) {
  return name.split('').toList().reversed == name;
}

extension StringUtilities on String {
  // Method to reverse a string
  String get reversed {
    return this.split('').reversed.join('');
  }

  // Method to check if the string is a palindrome
  bool get isPalindrome {
    return this.toLowerCase() == this.reversed.toLowerCase();
  }
}
